dpnp/backend/kernels/dpnp_krnl_logic.cpp

```diff
@@ -30,68 +30,89 @@
 #include "queue_sycl.hpp"
 
 
-template <typename _DataType, typename _ResultType>
+template <typename _DataType>
+class dpnp_all_c_kernel;
+
+template <typename _DataType>
 void dpnp_all_c(void* array1_in, void* result1, const size_t size)
 {
-    _DataType* array_in = reinterpret_cast<_DataType*>(array1_in);
-    _ResultType* result = reinterpret_cast<_ResultType*>(result1);
+    cl::sycl::event event;
+
+    const _DataType* array_in = reinterpret_cast<const _DataType*>(array1_in);
+    bool* result = reinterpret_cast<bool*>(result1);
 
-    bool res = true;
+    result[0] = true;
 
-    for (size_t i = 0; i < size; ++i)
+    if ((array1_in != nullptr) && (result1 != nullptr))
     {
-        if (!array_in[i])
-        {
-            res = false;
-            break;
-        }
-    }
+        cl::sycl::range<1> gws(size);
+        auto kernel_parallel_for_func = [=](cl::sycl::id<1> global_id) {
+            size_t i = global_id[0];
 
-    result[0] = res;
+            if (!array_in[i])
+            {
+                result[0] = false;
+            }
+        };
 
-    return;
+        auto kernel_func = [&](cl::sycl::handler& cgh) {
+            cgh.parallel_for<class dpnp_all_c_kernel<_DataType>>(gws, kernel_parallel_for_func);
+        };
+
+        event = DPNP_QUEUE.submit(kernel_func);
+    }
+
+    event.wait();
 }
 
-template <typename _DataType, typename _ResultType>
+template <typename _DataType>
+class dpnp_any_c_kernel;
+
+template <typename _DataType>
 void dpnp_any_c(const void* array1_in, void* result1, const size_t size)
 {
-    if ((array1_in == nullptr) || (result1 == nullptr))
-    {
-        return;
-    }
+    cl::sycl::event event;
 
     const _DataType* array_in = reinterpret_cast<const _DataType*>(array1_in);
-    _ResultType* result = reinterpret_cast<_ResultType*>(result1);
+    bool* result = reinterpret_cast<bool*>(result1);
 
-    bool res = false;
+    result[0] = false;
 
-    for (size_t i = 0; i < size; ++i)
+    if ((array1_in != nullptr) && (result1 != nullptr))
```

Comment:
What if `result1` is nullptr?
You initilizing before check - it is bad.

move nullptr check on top.

Reply:
"What if result1 is nullptr?"
Behavior unpredictable I think. The code will wait on uninitialized `event` by `event.wait()`